mgmt/rest/v2/config.go

```diff
@@ -33,17 +33,47 @@ type PolicyTable cpolicy.RuleTable
 
 type PolicyTableSlice []cpolicy.RuleTable
 
-// cdata.ConfigDataNode implements it's own UnmarshalJSON
+// PluginConfigItem represents the response of a plugin config items.
+//
+// swagger:response PluginConfigResponse
 type PluginConfigItem struct {
-	cdata.ConfigDataNode
+	// in: body
+	Config cdata.ConfigDataNode `json:"config"`
+}
+
+// PluginConfigParam type
+//
+//swagger:parameters setPluginConfigItem
+type PluginConfigParam struct {
+	// in: formData
+	Config string `json:"config"`
+}
+
+// PluginConfigDeleteParams defines parameters for deleting a config.
+//
+// swagger:parameters deletePluginConfigItem
+type PluginConfigDeleteParams struct {
+	// required: true
+	// in: path
+	PName string `json:"pname"`
+	// required: true
+	// in: path
+	PVersion int `json:"pversion"`
+	// required: true
+	// in: path
+	// enum: collector, processor, publisher
+	PType string `json:"ptype"`
+	// in: formData
+	// required: true
+	Config []string `json:"config"`
 }
 
 func (s *apiV2) getPluginConfigItem(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
 	var err error
 	styp := p.ByName("type")
 	if styp == "" {
 		cdn := s.configManager.GetPluginConfigDataNodeAll()
-		item := &PluginConfigItem{ConfigDataNode: cdn}
+		item := &PluginConfigItem{Config: cdn}
```

Comment:
Small suggestion: changing `cdn` to for example `cfg` 
Why: after changing the name ConfigDataNode to Config, `cdn` does not make sense as it had before.